src/citrine/_utils/batcher.py

```diff
@@ -0,0 +1,164 @@
+from abc import ABC, abstractmethod
+from collections import defaultdict
+from typing import List
+
+from citrine.resources.data_concepts import DataConcepts
+
+from gemd.util import writable_sort_order
+from gemd.entity.link_by_uid import LinkByUID
+
+
+class Batcher(ABC):
+    """Base class for Data Concepts batching routines."""
+
+    @abstractmethod
+    def batch(self, objects: List[DataConcepts], batch_size) -> List[List[DataConcepts]]:
+        """Collect a list of DataConcepts into batches according to some batching algorithm."""
+
+    @classmethod
+    def by_type(cls):
+        """Return a BatchByType batcher."""
+        return BatchByType()
+
+    @classmethod
+    def by_dependency(cls):
+        """Return a BatchByDependency batcher."""
+        return BatchByDependency()
```

Comment:
Could these be `@staticmethod` instead of class method? They don't rely on the `cls` type